### PR TITLE
Pre-Resource Policies

### PR DESCRIPTION
**Pull request by Bento007**
This is in preparation for resource policies

- adding allowed_policy_types to cloudNode, this controls what policy types can be attached to this node.
- changed statement to get/set_policy were appropriate
- added typing to groups and roles
- removed statement property for get_policy and set_policy functions
- removed policy property
- updated test cases
- Added a server side warning when a policy_type not on the allowed list is attached to a node.
- Simplified cloudNode.get_attributes
- Added cloudNode.get_info to return a clean representation of the node.
- Checking that a node exists before creating it's attached policy.
- Modified yaml to return allowed policies directly attached to cloudNode

### CODE REVIEW

File: fusillade/clouddirectory.py [hunk 1]
```diff
@@ -1183,45 +1192,48 @@ def _set_statement(self, statement: str):
...
         for attr in resp['Attributes']:
             self.__setattr__('_' + attr['Key']['Name'], attr['Value'].popitem()[1])
 
-    def get_attributes(self, attributes: List[str]):
-        attrs = dict()
-        if not attributes:
-            return attrs
+    def get_attributes(self, attributes: List[str]) -> Dict[str, str]:
+        """
+        retrieve the attribute values.
```
Comment (DailyDreaming) [inline fusillade/clouddirectory.py]: Nit: Not sure that this docstring is necessary.

File: fusillade/clouddirectory.py [hunk 2]
```diff
@@ -1140,40 +1125,64 @@ def create_policy(self, statement: str) -> str:
...
-        try:
-            if not self.policy:
-                self.create_policy(statement)
+    def set_policy(self, statement: str, policy_type: str = 'IAMPolicy'):
+        if policy_type in self.allowed_policy_types:
+            try:
+                # check if this object exists
+                self.cd.get_object_information(self.object_ref)
+            except cd_client.exceptions.ResourceNotFoundException:
+                raise FusilladeHTTPException(status=404, title="Not Found", detail="Resource does not exist.")
```
Comment (DailyDreaming) [inline fusillade/clouddirectory.py]: Nit: Consider using `requests.codes.not_found` instead of `404`.

Comment (Bento007) [reply fusillade/clouddirectory.py]: will fix in the future

File: fusillade/clouddirectory.py (hunk 1 above)
Comment (Bento007) [reply fusillade/clouddirectory.py]: not fixing at this time

Comment (Bento007) [reply fusillade/clouddirectory.py]: but i agree, it's  not needed

**Pull request merged**